homeassistant/components/trafikverket_camera/camera.py

```diff
@@ -0,0 +1,94 @@
+"""Camera for the Trafikverket Camera integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONF_LOCATION, DOMAIN
+from .coordinator import TVDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Trafikverket Camera."""
+
+    coordinator: TVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            TVCamera(
+                coordinator,
+                entry.title,
+                entry.data[CONF_LOCATION],
+                entry.entry_id,
+            )
+        ],
+    )
+
+
+class TVCamera(CoordinatorEntity[TVDataUpdateCoordinator], Camera):
+    """Implement Trafikverket camera."""
+
+    _attr_has_entity_name = True
+    coordinator: TVDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: TVDataUpdateCoordinator,
+        name: str,
+        location: str,
+        entry_id: str,
+    ) -> None:
+        """Initialize the camera."""
+        super().__init__(coordinator)
+        Camera.__init__(self)
+        self._attr_unique_id = entry_id
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.0",
```

Comment:
Model is optional, thus is it really necessary to set this to v1.0? Does the Trafikverket service have multiple versions?

Reply:
I set it here to refer to which api schema it refers to. Maybe pointless for someone else than me.
Each endpoint has a different schema so if you look at the other `trafikverket` integrations you see different models e.g. schemas.